modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheOffheapManager.java

```diff
@@ -1108,7 +1110,28 @@ private Metas getOrAllocateCacheMetas() throws IgniteCheckedException {
         try {
             int cleared = 0;
 
+            // Use random shift to reduce contention.
+            int shift = ThreadLocalRandom.current().nextInt(F.size(cacheDataStores().iterator()));
+
+            int cnt = 0;
+            for (CacheDataStore store : cacheDataStores()) {
```

Comment:
As far as I understood, the logic here is to split cacheDataStores() into 2 parts at random index and after that process right sublist before the left sublist. Can this change be simplified to minimize code duplication? For example, construct a new list from 2 sublists and process it as before.